File: tapak_asih_flutter/lib/config/tapak_asih_config.dart
/// Configuration class for TapakAsih Flutter SDK
class TapakAsihConfig {
  final String developerToken;
  final bool enableDebugLogs;
  final bool enableOfflineQueue;
  final int retryAttempts;

  TapakAsihConfig({
    required this.developerToken,
    this.enableDebugLogs = false,
    this.enableOfflineQueue = true,
    this.retryAttempts = 3,
  });

  /// Create configuration from map
  factory TapakAsihConfig.fromMap(Map<String, dynamic> map) {
    return TapakAsihConfig(
      developerToken: map['developerToken'] as String,
      enableDebugLogs: map['enableDebugLogs'] as bool? ?? false,
      enableOfflineQueue: map['enableOfflineQueue'] as bool? ?? true,
      retryAttempts: map['retryAttempts'] as int? ?? 3,
    );
  }

  /// Convert configuration to map
  Map<String, dynamic> toMap() {
    return {
      'developerToken': developerToken,
      'enableDebugLogs': enableDebugLogs,
      'enableOfflineQueue': enableOfflineQueue,
      'retryAttempts': retryAttempts,
    };
  }
}

File: src/main/java/com/paondev/lib/tapakasih/config/TapakAsihConfig.java
package com.paondev.lib.tapakasih.config;

/**
 * Configuration class for TapakAsih SDK
 */
public class TapakAsihConfig {
    private String developerToken;
    private boolean enableDebugLogs;
    private boolean enableOfflineQueue;
    private int retryAttempts;
    
    private TapakAsihConfig(Builder builder) {
        this.developerToken = builder.developerToken;
        this.enableDebugLogs = builder.enableDebugLogs;
        this.enableOfflineQueue = builder.enableOfflineQueue;
        this.retryAttempts = builder.retryAttempts;
    }
    
    public String getDeveloperToken() {
        return developerToken;
    }
    
    public boolean isEnableDebugLogs() {
        return enableDebugLogs;
    }
    
    public boolean isEnableOfflineQueue() {
        return enableOfflineQueue;
    }
    
    public int getRetryAttempts() {
        return retryAttempts;
    }
    
    /**
     * Builder for TapakAsihConfig
     */
    public static class Builder {
        private String developerToken;
        private boolean enableDebugLogs = false;
        private boolean enableOfflineQueue = true;
        private int retryAttempts = Constants.MAX_RETRY_ATTEMPTS;
        
        public Builder(String developerToken) {
            if (developerToken == null || developerToken.trim().isEmpty()) {
                throw new IllegalArgumentException("Developer token cannot be null or empty");
            }
            this.developerToken = developerToken;
        }
        
        public Builder setEnableDebugLogs(boolean enableDebugLogs) {
            this.enableDebugLogs = enableDebugLogs;
            return this;
        }
        
        public Builder setEnableOfflineQueue(boolean enableOfflineQueue) {
            this.enableOfflineQueue = enableOfflineQueue;
            return this;
        }
        
        public Builder setRetryAttempts(int retryAttempts) {
            this.retryAttempts = retryAttempts;
            return this;
        }
        
        public TapakAsihConfig build() {
            return new TapakAsihConfig(this);
        }
    }
}